src/main/java/uk/gov/hmcts/reform/migration/query/ElasticSearchQuery.java

```diff
@@ -8,11 +8,40 @@ public class ElasticSearchQuery {
     private static final String START_QUERY = """
         {
           "query": {
-            "match_all": {}
+            "bool": {
+               "must_not": {
+                 "exists": {
```

Comment:
What if these fields change while this migration is running?

If 'caseHandedOffToLegacySite' is set (or the state of the case changes to no longer match) then a case would no longer be returned by Elasticsearch.

So the following sequence of events is possible:

1. Case is found in ES query and stored in list of cases to migrate
2. While migration continues running the case is updated by something else in such a way that it would no longer be returned by the ES search query in this tool
3. Migration continues running on a case that it should not be migrating

If our goal is 'only migrate cases matching these criteria' then we cannot rely solely on Elasticsearch to do this - to be correct any filtering logic encoded in the Elasticsearch query would need to be reapplied against the case data from the start event, since only that version of the case data is guaranteed to be current.

Reply:
a null checking on caseHandedOffToLegacySite is added before updating the case, ok?

Reply:
Could the case move to a state where it should not be processed?

Reply:
any state change not affect the data migration

Reply:
Fine so long as the mutation of these fields does not cause a case to be migrated that should not be